src/client/Client.cc

```diff
@@ -11509,15 +11509,43 @@ int Client::_setxattr(Inode *in, const char *name, const void *value,
     if (vxattr) {
       if (vxattr->readonly)
 	return -EOPNOTSUPP;
-      if (vxattr->name.compare(0, 10, "ceph.quota") == 0 && value)
-	check_realm = true;
+      if ((vxattr->name.compare(0, 10, "ceph.quota") == 0 ||
+          vxattr->name.compare(0, 15, "ceph.user_quota") == 0 ||
+          vxattr->name.compare(0, 16, "ceph.group_quota") == 0) && value)
+        check_realm = true;
+
+      string sname(name);
+      if(sname.find("@") != sname.npos){
+        char buf[256];
+        string sub_name = sname.substr(0, sname.find("@"));
+        if (sub_name == "ceph.user_quota.max_bytes") {
+          string cp = sname.substr(sname.find("@")+1);
+          struct passwd *pw = getpwnam(cp.c_str());
+          if(!pw)
+            return -EINVAL;
+          uid_t uid = pw->pw_uid;
+          snprintf(buf, sizeof(buf),"%s@%lld", sub_name.c_str(), (long long int)uid);
+        } else if(sub_name == "ceph.group_quota.max_bytes") {
+          string cp = sname.substr(sname.find("@")+1);
+          struct group *data = getgrnam(cp.c_str());
```

Comment:
getting uid/gid from /etc/passwd (on the host that runs mds) is wrong.  cephfs client hosts and mds host do not have the same name->uid/gid mapping.

Reply:
Thanks for your reply. In the setfattr and getfattr process,we get uid/gid from /etc/passwd on the client host and pass them to the mds host, mds host does not do mapping operation.